miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt

```diff
@@ -159,14 +174,32 @@ abstract class MiniAppMessageBridge {
 
     private fun onRequestPermission(callbackObj: CallbackObj) {
         try {
-            val permissionParam = Gson().fromJson<Permission>(
+            val permissionParam = Gson().fromJson<DevicePermission>(
                 callbackObj.param.toString(),
-                object : TypeToken<Permission>() {}.type
+                object : TypeToken<DevicePermission>() {}.type
             )
 
             requestPermission(
                 MiniAppPermissionType.getValue(permissionParam.permission)
-            ) { isGranted -> onRequestPermissionsResult(
+            ) { isGranted -> onRequestDevicePermissionsResult(
+                callbackId = callbackObj.id,
+                isGranted = isGranted
+            ) }
+        } catch (e: Exception) {
+            bridgeExecutor.postError(callbackObj.id, "${ErrorBridgeMessage.ERR_REQ_PERMISSION} ${e.message}")
+        }
+    }
+
+    private fun onRequestDevicePermission(callbackObj: CallbackObj) {
+        try {
+            val permissionParam = Gson().fromJson<DevicePermission>(
+                callbackObj.param.toString(),
+                object : TypeToken<DevicePermission>() {}.type
+            )
+
+            requestDevicePermission(
```

Comment:
I think you'll need to do something here similar to what you did when changing the `getUserName` functions. First try calling `requestDevicePermission` and then if it's not implemented, call the deprecated `requestPermission` instead.

Reply:
Thanks, I have updated this part.